assets/src/edit-story/app/font/actions/useLoadFontFiles.js

```diff
@@ -25,41 +25,113 @@ import { useCallback } from 'react';
-    if (!family || service !== 'fonts.google.com') {
-      return;
-    }
+  const maybeEnqueueFontStyle = useCallback(
+    async (
+      {
+        font: { family, service, variants },
+        fontWeight,
+        fontStyle,
+        fontSize,
+      } = { fontSize: 0 }
+    ) => {
+      if (
+        !family ||
+        !fontWeight ||
+        !fontStyle ||
+        service !== 'fonts.google.com'
+      ) {
+        return null;
+      }
+
+      const handle = cleanForSlug(family);
+      const elementId = `${handle}-css`;
+      const fontFaceSet = `${fontStyle} ${fontWeight} ${fontSize}px '${family}'`;
 
-    const handle = cleanForSlug(family);
-    const id = `${handle}-css`;
-    const element = document.getElementById(id);
+      const hasFontLink = () => {
+        return document.getElementById(elementId);
+      };
 
-    if (element) {
-      return;
-    }
+      const appendFontLink = () => {
+        return new Promise((resolve, reject) => {
+          const src = getGoogleFontURL([{ family, variants }]);
+          const fontStylesheet = document.createElement('link');
+          const fontHref = src.replace('display=swap', 'display=auto');
+          fontStylesheet.id = elementId;
+          fontStylesheet.href = fontHref;
+          fontStylesheet.rel = 'stylesheet';
+          fontStylesheet.type = 'text/css';
+          fontStylesheet.media = 'all';
+          fontStylesheet.crossOrigin = 'anonymous';
+          fontStylesheet.addEventListener('load', () => resolve());
+          fontStylesheet.addEventListener('error', (e) => reject(e));
+          document.head.appendChild(fontStylesheet);
+        });
+      };
 
-    const src = getGoogleFontURL([{ family, variants }]);
+      const ensureFontLoaded = async () => {
+        if (document?.fonts) {
+          await document.fonts.load(fontFaceSet);
+          return document.fonts.check(fontFaceSet);
+        } else {
+          return null;
+        }
+      };
 
-    const fontStylesheet = document.createElement('link');
-    fontStylesheet.id = id;
-    fontStylesheet.href = src;
-    fontStylesheet.rel = 'stylesheet';
-    fontStylesheet.type = 'text/css';
-    fontStylesheet.media = 'all';
-    fontStylesheet.crossOrigin = 'anonymous';
+      if (!hasFontLink()) {
+        await appendFontLink();
+      }
 
-    document.head.appendChild(fontStylesheet);
-  }, []);
+      return ensureFontLoaded();
+    },
+    []
+  );
+
+  /**
+   * It allows control each text element font-face for multiple types and font family aspect
+   *
+   * @param {string} aspect Font family aspect that should be synced between one or more elements
+   * @param {Object} state Font family state to be synced with `maybeEnqueueFontStyle`
+   * @param {Array} elements List of elements selected to be processed
+   * @return {Promise} Returns a Promise after process all font with `maybeEnqueueFontStyle`
+   */
+  const ensureFontFaceSetIsAvailable = (aspect, state, elements) => {
```

Comment:
I think it'd be better if we here do `maybeEnqueueFontStyleSet` and just provide an array of args for `maybeEnqueueFontStyle` and call it directly. That would be the most flexible, and can be optimized as needed. At the same time, we can easily do `selectedEleemnts.map(...)` at the call site to construct such an array.